### DISCUSSION TITLE
How to get into analytics engineering from scratch?

### DISCUSSION THREAD

**Discussion by gwenwindflower** [dbt-labs/docs.getdbt.com#1540] [Archive]
We have great resources on learning dbt if you're SQL familiar, but what about if you're pivoting careers, or coming from a heavily spreadsheet-, Looker-, or Tableau-based background? I want to start mapping out a guide to the best ways to learn SQL, then dbt, and then establish a portfolio to break into the industry. Please suggest resources and we'll shape this up over time, eventually this can live in a Guide on the Devhub.

## 1. Learn or sharpen SQL skills.

Tons of great free options for this:
  - [Mode's SQL Bootcamp](https://mode.com/sql-tutorial/)
  - [Select * SQL book](https://selectstarsql.com/)
  - [SQLBolt Interactive SQL Lessons](https://sqlbolt.com/)

## 2. Learn dbt.
1. Set up a project in dbt Cloud with a free account. I recommend using BigQuery as your warehouse because they have a generous free tier and lots of cool public datasets you can model with! Work through the [Getting Started Guide](https://docs.getdbt.com/guides/getting-started).
2. Next, work through the free courses from [dbt Labs](https://courses.getdbt.com/collections) and apply what you're learning to an interesting dataset you find in BigQuery.
3. Join the [dbt Slack community](https://www.getdbt.com/community/join-the-community/) and share your work in your repo, ask and answer questions.
4. To be continued...

If you feel like you'd do better with a paid course, co:rise's course taught by Emily Hawkins and Analytics Engineers Club from Claire Carrol and Michael Kaminsky are both excellent options that I recommend highly.

**Comment by lgreene27** [comment]
Love this! One of the harder things to learn is not the actual syntax of SQL, but the practice of data modeling and structure to which you apply that syntax. I think a great way to learn here is from others that have done us all the favor of learning in public - my favorite here is [Gitlab](https://gitlab.com/gitlab-data/analytics/-/tree/master/transform/snowflake-dbt/models) who shares almost their entire repo so you can see how they use sql in practice. And, the age old advice, see if you can work on real projects - hook up with a nonprofit or political cause you care about and build them out a data stack or find a live public dataset that is interesting to you to build on top of. Would love to know what other resources or suggestions people have for helping folks to get hands on beyond the simple demo project setups

**Reply by brian-gillet** [reply]
+1 @lgreene27  - I was just thinking someone who is fluent in Tableau might want to work backwards from what they are familiar with to what they are most unfamiliar with to become an Analytics Engineer. 

Once they pick up some SQL and can load some data into a data warehouse (What resources exist to help someone with loading data into a data warehouse?) they can model some data and then visualize it in Tableau to see the whole process come to fruition. 

Lucky for Tableau developers, they have a vibrant community that hosts many different viz or data prep challenges. Here are a few challenges where they could source data and solve for a business problem: 

[Preppin' Data](https://preppindata.blogspot.com/p/the-challenge-index.html) - data prep challenges with data sets
[Makeover Monday](https://www.makeovermonday.co.uk/data/) - data viz challenges with data sets
[Workout Wednesday](https://www.workout-wednesday.com/) - data viz challenges with data sets

There's a preppin' data challenge with a multi-table data set I'd recommend for beginners who need to learn to do joins and some data clean up available [here](https://preppindata.blogspot.com/2022/02/2022-week-7-call-center-agent-metrics.html)

**Reply by gwenwindflower** [reply]
great idea @brian-gillet this is making me think we could offer some alternate starting points that converge towards dbt

**Reply by gwenwindflower** [reply]
100% @lgreene27 that's actually exactly where i want to go with Step 3 -- i just didn't have time to start compiling projects to volunteer on, but that's been a hobby horse of @jasnonaz and myself for a long time now and i absolutely want that to be the end goal here!

**Comment by SamHarting** [comment]
**A Shot For Shot Story of My Transition Into Analytics Engineering**

When I say I didn't know what Analytics Engineering was until a few weeks into my first Analytics Engineering job I genuinely mean it. So how did I end up there? Well, I am glad you asked.

My first career, the one for which I studied in college, was accounting. I passed my four CPA exams and did tax work. I honestly didn't mind the work that much. I even found it interesting. However, the career of a CPA is very structured. Generally, if you aren't moving forward in your career then you are moving out of your career. The future of my life looked a lot less like tax work and a lot more like managing clients, making sales, and managing a team. Some of these tasks may appeal to me eventually, but they were moving at me quickly and I was not excited about it. Not to mention, the type of tax work I was doing was getting quite complicated. Not the fun "lets tackle this problem and find a rewarding solution" kind of complicated, the "what in the world was Congress thinking when they made this law" kind of complicated. That, combined with the anxiety of the pandemic and long hours during the winter and spring, convinced me it was time to go.

But what do I want to do? And where do I start?

At my CPA firm I had the opportunity to start working with a product called Alteryx. We were going to use it to do some basic functions on excel sheets that clients provided for us. My team was tasked with learning the software and figuring out how to make it useful in our organization. I started the tutorials and found that I was really enjoying the work. I had done some programming (C#) in college as well as a VERY small amount of SQL. I thought to myself, "man it would be great if my job were more like this".... so I started seeing if it could! We had a data team in our firm. They used PowerBI to create dashboards for our clients.

So at this point, my interest in data has been piqued, that is likely where **you, the reader** are at. So here is my first piece of actionable advice:

_**Talk to people in the data industry. Anyone. About anything.**_

I went to our data team leader and asked to schedule a meeting. When it came time I asked him incredibly basic questions:

1. What do you actually do?
2. What skills do you utilize?
3. What skills of mine are useful for this job?
4. What would it take for me to transition to your team?

Your questions might look a bit different but explore the curiosity as much as possible! Don't have anyone in your life that works in data? Join the dbt Community Slack and ask if anybody would be willing to have a thirty minute zoom with you. You might get a data practitioner from a big company, you might get someone from dbt itself!

I also reached out to current and former members of dbt because I had a connection here. That part, admittedly, was very lucky. I gleaned a lot from those conversations but the first step was clear:

_**Learn SQL, NOW!**_

And guess what y'all, there are So. Many. Ways. to learn SQL. My path was Udemy, others have used Codecademy, others Mode (we did the Mode courses internally as part of the apprenticeship program when I got here). Some of these options are paid, some of these options are not. Want to make sure you follow through on learning by signing up for a class? I guarantee you that exists. A local college could have a class or you could even complete an entire bootcamp. Of the four apprentices (two software engineering, two analytics engineering) I was the only one that hadn't completed a bootcamp. So... do you need to do one? No. Does it increase your chances of success? Based off of that very small sample size, it might! It might not!

At this point I applied for the apprenticeship. I had completed the majority of my SQL class and now it was time to read more on the company. I honestly read mostly what was going on at dbt. Some of it meant something to me, some of it didn't. This step was vital but I am going to be honest and say I think I could have done this differently. Did I end up successful anyways? Yes. But let me give you better advice than what I gave myself.

_**Read about dbt yes, but read about the data industry as a whole too. Follow your curiosity. When you stumble upon something that doesn't make sense, look it up. Don't spin your wheels forever, move on when you need to. But let your interest take you across the spectrum of data.**_

I will say, specifically, you should absolutely learn about the modern data stack and ELT processes. Literally just type those buzzwords into Google (or Bing if you nasty) and start reading. Watch videos if that is your preferred method of learning. That is something I did not do that I absolutely should have, videos keep my attention way more than articles.

At this point, you should know some SQL and have an idea of what is going on in the data world these days. If your goal is really to be an analytics engineer then dbt is the place to go:

_**Take the dbt Fundamentals course online, it is free!**_

There are other courses on there too, check them out! If something doesn't make sense, redo it. Still doesn't make sense? Read about it. Still doesn't make sense? Reach out to the dbt Slack or the contacts you made previously. 

Now the next step I had already done, but this is probably the appropriate timeline:

_**Apply for jobs!**_

Specifically if you can find apprenticeships or internships that will work best, but there might be some opportunities for you to go straight onto a team with the skills you already have. More and more companies are realizing there is a talent shortage and are creating apprenticeships. I believe there are even organizations that help place you into an apprenticeship! I have not used them, so I am hesitant to endorse or list them, but it is worth exploring.

**BONUS ROUND**
The tech world moves fast, we know this. Even since I have started my career in September of 2021, it is looking like Python will be a necessity rather than a way to stand out. Look at the same places you learned SQL for Python courses and start learning!

**Reply by wasilaq** [reply]
I think talking to others in the industry is a great tip. We have a lot of easy avenues for this already. Attend meetup events and get involved in the community Slack.

**Comment by wasilaq** [comment]
Love this idea. I listed some more sections that I think we should add.

**What's analytics engineering and why does it matter?**
- [The Modern Data Stack: Past, Present, and Future](https://www.getdbt.com/blog/future-of-the-modern-data-stack/)
- [What is analytics engineering?](https://www.getdbt.com/what-is-analytics-engineering/)

**Git & Github**
- [Git, GitHub, & GitHub Desktop for beginners video](https://www.youtube.com/watch?v=8Dd7KRpKeaE)
- [Atlassian's tutorials](https://www.atlassian.com/git/tutorials) as a reference.

**CLI**
This section assumes that being able to use dbt Core is necessary for breaking into the industry, which I think is true but I might be totally wrong there.

- [Introduction to the Command Line](https://launchschool.com/books/command_line)
- [What is an IDE?](https://www.freecodecamp.org/news/what-is-an-ide-in-programming-an-ide-definition-for-developers/)
- [VS Code](https://code.visualstudio.com/)

Re: SQL, I also found [this article](https://learnsql.com/blog/what-is-common-table-expression/) on CTE's today.

**Reply by gwenwindflower** [reply]
yessss this is 🔥

**Reply by SamHarting** [reply]
My other piece of advice is to do an apprenticeship with Wasila. She sent me so many good links.

**Comment by ghost** [comment]
The amount of times and places I've shared this with people is getting ridiculous. 😅
